services/src/main/java/org/keycloak/protocol/oidc/OIDCAdvancedConfigWrapper.java

```diff
@@ -321,6 +321,17 @@ public String getFrontChannelLogoutUrl() {
         return getAttribute(OIDCConfigAttributes.FRONT_CHANNEL_LOGOUT_URI);
     }
 
+    public boolean isFrontChannelLogoutSessionRequired() {
+        String frontChannelLogoutSessionRequired = getAttribute(OIDCConfigAttributes.FRONT_CHANNEL_LOGOUT_SESSION_REQUIRED);
+        // Include session by default for backwards compatibility
+        return frontChannelLogoutSessionRequired == null ? true : Boolean.parseBoolean(frontChannelLogoutSessionRequired);
```

Comment:
Accordin to https://openid.net/specs/openid-connect-frontchannel-1_0.html#RPLogout ,
It should be false if frontChannelLogoutSessionRequired is not defined.
WDYT?

Reply:
Yes, however another point is backwards compatibility. The previous version of Keycloak already has support for "FrontChannelLogout", but in fact it uses "frontChannelLogoutSessionRequired" set to true as it included "sid" and "iss" claims. Previous versions of Keycloak doesn't have this attribute defined, so switching this to false by default will in fact be broken backwards compatibility behaviour for them. Unless we add hook to the migration, which itself is typically a painful and can have other side-effects on performance etc...

So I suggest to keep this to true for backwards compatibility, but use default value "false" for the case when OIDC client is created or updated through OIDC dynamic client registration. In this case, it is fine to use "false" by default as we won't break backwards compatibility. And at the same time, we will be fine with the OIDC specification and use "false" for OIDC registered client. WDYT?

Reply:
Yes, I agree with your proposal.